python/ray/data/_internal/execution/operators/input_data_buffer.py

```diff
@@ -48,6 +48,8 @@ def start(self, options: ExecutionOptions) -> None:
             self._input_data = self._input_data_factory()
             self._is_input_initialized = True
             self._initialize_metadata()
+        for bundle in self._input_data:
```

Comment:
add a comment: Other operators usually record the input metrics in `add_input`. `InputDataBuffer` doesn't allow adding inputs, so we record the input metrics here.